test/unit_test/instantiate_fft_tests.hpp

```diff
@@ -211,18 +220,107 @@ INSTANTIATE_TEST_SUITE_P(OffsetsMDErrorRegressionTest, FFTTest,
+                             ::testing::Values(sizes_t{9}, sizes_t{16}, sizes_t{64}, sizes_t{512}, sizes_t{4096},
+                                               sizes_t{16, 512}),
+                             ::testing::Values(1.0), scales)),
+                         test_params_print());
+
+INSTANTIATE_TEST_SUITE_P(workItemStridedOOPInOrder, FFTTest,
+                         ::testing::ConvertGenerator<layout_param_tuple>(::testing::Combine(
+                             oop_unpacked_unpacked_layout, both_directions, complex_storages,
+                             ::testing::Values(1, 3, 33000ul),
+                             ::testing::Values(layout_params{{3}, {4}, {7}}, layout_params{{8}, {11}, {2}},
+                                               layout_params{{9}, {3}, {4}, 30, 40}))),
+                         test_params_print());
+// The LikeBatchInterleaved tests must have stride >= number of transforms
+INSTANTIATE_TEST_SUITE_P(
+    workItemStridedOOPLikeBatchInterleaved, FFTTest,
+    ::testing::ConvertGenerator<layout_param_tuple>(::testing::Combine(
+        oop_unpacked_unpacked_layout, both_directions, complex_storages, ::testing::Values(1, 10, 33),
+        ::testing::Values(layout_params{{8}, {33}, {99}, 1, 3}, layout_params{{8}, {33}, {2}, 1, 16},
+                          layout_params{{8}, {2}, {66}, 16, 2}))),
+    test_params_print());
+INSTANTIATE_TEST_SUITE_P(workItemStridedIP, FFTTest,
+                         ::testing::ConvertGenerator<layout_param_tuple>(::testing::Combine(
+                             ip_unpacked_unpacked_layout, both_directions, complex_storages,
+                             ::testing::Values(1, 3, 33000ul),
+                             ::testing::Values(layout_params{{3}, {4}, {4}}, layout_params{{9}, {3}, {3}, 25, 25}))),
+                         test_params_print());
+INSTANTIATE_TEST_SUITE_P(
+    workItemStridedLikeBatchInterleaved, FFTTest,
+    ::testing::ConvertGenerator<layout_param_tuple>(::testing::Combine(
+        ip_unpacked_unpacked_layout, both_directions, complex_storages, ::testing::Values(1, 3, 33),
+        ::testing::Values(layout_params{{3}, {66}, {66}, 2, 2}, layout_params{{6}, {40}, {40}, 1, 1}))),
+    test_params_print());
+
+// clang-format off
+// Arbitrary interleaved FFT test suites
+// The strides and distances are set so that no elements overlap but there are no single continuous dimension in memory either.
+// This configuration is impractical but technically valid. For instance for n_batches=4, fft_size=4, stride=4, distance=3:
+// Index in memory:     0    1    2    3    4    5    6    7    8    9    10   11   12   13   14   15   16   17   18   19   20   21
+// Batch and FFT index: b0i0           b1i0 b0i1      b2i0 b1i1 b0i2 b3i0 b2i1 b1i2 b0i3 b3i1 b2i2 b1i3      b3i2 b2i3           b3i3
+// clang-format on
+INSTANTIATE_TEST_SUITE_P(workItemStridedArbitraryInterleaved, FFTTest,
```

Comment:
Do you have tests to make sure that we allow configurations that read from the same index? Similar to `OverlapRead` here https://github.com/codeplaysoftware/portFFT/blob/romain/stride_distance/test/unit_test/instantiate_fft_tests.hpp#L224

Reply:
We agreed separately that we won't support that right now because of the test setup not supporting it.
Currently I have these tests for overlap
https://github.com/codeplaysoftware/portFFT/blob/fd9d7ccc5d6ec37fe673210c00a58ec5a159b7ff/test/unit_test/instantiate_fft_tests.hpp#L313-L318
the first one has an overlap in the forward domain
```
C++ exception with description "Domain forward: batch 1 collides with first batch at index 1" thrown in the test body.
```
the second has an overlap in the backward domain
```
C++ exception with description "Domain backward: batch 2 collides with first batch at index 6" thrown in the test body.
```
I'll update the readme too

Reply:
Leaving this open to clarify in the readme that overlap reads are not supported